doc/finalization.md

```diff
@@ -8,7 +8,9 @@ provide more efficient memory management, optimizations, improved execution, or
 other benefits.
 
 In general, it is best to use basic finalizers whenever possible (eg. when
-access to JavaScript is _not_ needed).
+access to JavaScript is _not_ needed). To ensure that all finalizers are basic
```

Comment:
Let's make this passive voice: The `NODE_ADDON_API_REQUIRE_BASIC_FINALIZERS` preprocessor directive can be defined to ensure that all finalizers are basic.

Reply:
Addressed in https://github.com/nodejs/node-addon-api/pull/1568/commits/23b57cc93946df51f30cbfada61213be14f61421